### ISSUE TITLE
Regression in 0.28

### ISSUE THREAD

**Issue by atti92** [mkdocstrings/griffe#160]
**Describe the bug**
Using mkdocs The documentation generation fail with 0.28, but succeeds with 0.27.5

**To Reproduce**
I don't have the package or any of it's dependencies installed in my environment, only mkdocs related packages:
```
mkdocs==1.4.3
mkdocs-autorefs==0.4.1
mkdocs-awesome-pages-plugin==2.9.1
mkdocs-git-revision-date-plugin==0.3.2
mkdocs-macros-plugin==0.7.0
mkdocs-material==9.1.12
mkdocs-material-extensions==1.1.1
mkdocs-render-swagger-plugin==0.0.4
mkdocstrings[python]==0.21.2
mkdocs-gen-files==0.5.0
mkdocs-literate-nav==0.6.0
```
I have a project structure as follows:
- docs/
   - dist/index.md  contents: `::: mypackage.mymodule.mysubmodule`
   - mkdocs.yml
- src/mypackage ...

**Expected behavior**
It should still work.

**Screenshots**

**System (please complete the following information):**
- `griffe` version: 0.28.0
- Python version: 3.8
- OS: Linux

**Additional context**
My package is quite big, and I cannot share it.

**Comment by pawamoy** [comment]
Thanks for the report? What is the actual error?

**Comment by atti92** [comment]
```
ERROR    -  mkdocstrings: mypackage.mymodule.mysubmodule could not be found
ERROR    -  Error reading page 'reference/mymodule/mysubmodule .md':
ERROR    -  Could not collect 'mypackage.mymodule.mysubmodule'

Aborted with a BuildError!
```

**Comment by atti92** [comment]
I spent some time commenting/uncommenting some code, and found the problematic lines:
```
class MyModel(pydantic.BaseSettings):
    pass

    if Path("/path/to/secret").exists():
        class Config:
            secrets_dir = "/path/to/secret"
```

**Comment by pawamoy** [comment]
Thanks for the update.

A type annotation of `bool | False` seems incorrect to me. `bool | Literal[False]` would be better, but is semantically redundant with just `bool`.

For your second code sample, I don't understand why this would trigger an error :confused: 
Could you send the verbose logs by running `mkdocs build -v`?

**Comment by pawamoy** [comment]
I was eventually able to replicate :slightly_smiling_face:  Working on a fix.

**Comment by pawamoy** [comment]
Should be fixed in 0.28.1 :slightly_smiling_face:
